app/controllers/dashboard/form/base_controller.rb

```diff
@@ -73,11 +78,40 @@ def cancel_path
           end
         end
 
+        helper_method :allow_curation?
+        def allow_curation?
+          curation_eligible? && in_publish_edit_action?
+        end
+
+        helper_method :allow_publish?
+        def allow_publish?
+          if @resource.is_a?(Collection)
+            true
+          else
+            !@resource.draft_curation_requested
+          end
+        end
+
         helper_method :param_key
         def param_key
           resource_klass.model_name.param_key
         end
 
+        def in_publish_edit_action?
+          current_controller = params[:controller]
+          current_action = params[:action]
+
+          current_controller == 'dashboard/form/publish' && current_action == 'edit'
+        end
+
+        def data?
+          @resource.work.work_type == 'dataset' || @resource.work.work_type == 'software_or_program_code'
+        end
+
+        def curation_eligible?
+          !@resource.is_a?(Collection) && !@resource.draft_curation_requested && data?
+        end
```

Comment:
FYI `data?` is just a placeholder until the work for `deposit_pathway` is merged